### PR TITLE
assignment1

### PR DESCRIPTION
**Pull request by mingxguo**

### CODE REVIEW

File: mingxguo/assignment-1/anagrams_test.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "mingxguo/assignment-1/anagrams.h"
+#include "shared/testing/assertions.h"
+#include "shared/testing/test.h"
+
+namespace {
+
+TEST(WordAnagramsCaseSensitive) {
```
Comment (elitsa-codeu) [inline mingxguo/assignment-1/anagrams_test.cpp]: Thanks for adding tests, Ming!

One more useful example to make the test set complete for all functions would be to test for the corner case with empty strings.

File: mingxguo/assignment-1/anagrams_test.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	EXPECT_FALSE(AreWordAnagramsCaseSensitive("Listen", "Silent"));
+	EXPECT_TRUE(AreWordAnagramsCaseSensitive("elvis", "lives"));
+}
+TEST(WordAnagramsCaseInsensitive) {
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("LisTen", "siLenT"));
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("Listen", "Silent"));
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("elvis", "lives"));
+	EXPECT_FALSE(AreWordAnagramsCaseInsensitive("abc", "ab"));
+}
+TEST(SentenceAnagrams1) {
```
Comment (elitsa-codeu) [inline mingxguo/assignment-1/anagrams_test.cpp]: To test the behaviour of the function better it would be good to have at least one EXPECT_FALSE test too.

File: mingxguo/assignment-1/anagrams_test.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+TEST(WordAnagramsCaseSensitive) {
+	EXPECT_TRUE(AreWordAnagramsCaseSensitive("LisTen", "siLenT"));
+	EXPECT_FALSE(AreWordAnagramsCaseSensitive("Listen", "Silent"));
+	EXPECT_TRUE(AreWordAnagramsCaseSensitive("elvis", "lives"));
+}
+TEST(WordAnagramsCaseInsensitive) {
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("LisTen", "siLenT"));
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("Listen", "Silent"));
+	EXPECT_TRUE(AreWordAnagramsCaseInsensitive("elvis", "lives"));
+	EXPECT_FALSE(AreWordAnagramsCaseInsensitive("abc", "ab"));
```
Comment (elitsa-codeu) [inline mingxguo/assignment-1/anagrams_test.cpp]: Maybe add one more example, where the words are not anagrams and strings are with equal lengths.
